### DISCUSSION TITLE
Are the array fields advanced filters supported on Go SDK?

### DISCUSSION THREAD

**Discussion by juanandreas** [milvus-io/milvus#33965] [Q&A and General discussion]
The advanced filters described in this document: https://milvus.io/docs/array_data_type.md (`ARRAY_CONTAINS`, `ARRAY_CONTAINS_ALL`, `ARRAY_CONTAINS_ANY`, and `ARRAY_LENGTH`) require a `filter` field in the arguments of client.search or client.query. I don't see this filter argument in the search and query source code.

```
	Search(ctx context.Context, collName string, partitions []string,
		expr string, outputFields []string, vectors []entity.Vector, vectorField string, metricType entity.MetricType, topK int, sp entity.SearchParam, opts ...SearchQueryOptionFunc) ([]SearchResult, error)

	Query(ctx context.Context, collectionName string, partitionNames []string, expr string, outputFields []string, opts ...SearchQueryOptionFunc) (ResultSet, error)
```

I see we can add them as part of `expr` but they do not work as intended. e.g `ARRAY_CONTAINS_ANY(x, y)` doesn't return true for rows where some elements in `x` exist in `y`. Though it works when `y` only has a single element.

**Comment by yhmo** [comment]
expr is a string, all the SDKs use the same expression to do filtering.
The example of ARRAY_CONTAINS_ANY: https://milvus.io/docs/array_data_type.md#Advanced-filtering

Assume there is an Array field named "color_coord" in a collection.
A row is inserted as: {"id": xxx, "vector": [xxxx], "color_coord": [1, 2, 3, 4, 5]}
The expr should be expr="ARRAY_CONTAINS_ANY(color_coord, [1, 3, 5])"
